### PR TITLE
VZ-5771 rancher backup operator onboarding

### PR DESCRIPTION
**Pull request by theabmitra**
PR for onboarding rancher backup operator as a component in Verrazzano

### CODE REVIEW

File: platform-operator/controllers/verrazzano/component/rancherbackup/rancher_backup_operator_component.go [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+	comp := effectiveCR.Spec.Components.RancherBackup
+	if comp == nil || comp.Enabled == nil {
+		return false
+	}
+	return *comp.Enabled
+}
+
+// IsInstalled returns true only if Rancher Backup is installed on the system
+func (rb rancherBackupHelmComponent) IsInstalled(ctx spi.ComponentContext) (bool, error) {
+	for _, nsn := range deployments {
```
Comment (michael-cico) [inline platform-operator/controllers/verrazzano/component/rancherbackup/rancher_backup_operator_component.go:81]: Why not just rely on the `HelmComponent.IsInstalled`?

File: platform-operator/controllers/verrazzano/component/rancherbackup/rancher_backup.go [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+	imagePullSecretHelmKey = "image.imagePullSecrets[0]"
+)
+
+// isRancherBackupOperatorReady checks if the Rancher Backup deployment is ready
+func isRancherBackupOperatorReady(context spi.ComponentContext) bool {
+	return status.DeploymentsAreReady(context.Log(), context.Client(), deployments, 1, componentPrefix)
+}
+
+// AppendOverrides appends Helm value overrides for the Rancher Backup component's Helm chart
+func AppendOverrides(compContext spi.ComponentContext, _ string, _ string, _ string, kvs []bom.KeyValue) ([]bom.KeyValue, error) {
```
Comment (michael-cico) [inline platform-operator/controllers/verrazzano/component/rancherbackup/rancher_backup.go:28]: looks like this is a no-op?  you can probably get rid of this func if that's the case.

File: platform-operator/thirdparty/charts/README.md [hunk 3]
```diff
@@ -135,10 +135,23 @@ The `verrazzano-application-operator` folder was created manually.
...
 helm fetch vmware-tanzu/velero --untar=true --version=${VELERO_HELM_CHART_VERSION}
+```
+
+### Rancher Backup Operator
+
+The `rancher-charts` folder was created by running the following commands:
+
+```shell
+helm repo add rancher-charts https://charts.rancher.io
+rm -rf rancher-backup-crd
```
Comment (michael-cico) [inline platform-operator/thirdparty/charts/README.md:152]: I this should be `rm -rf rancher-backup/crds` ?

Comment (theabmitra) [reply platform-operator/thirdparty/charts/README.md:152]: NA

File: platform-operator/controllers/verrazzano/component/rancherbackup/rancher_backup.go (hunk 2 above)
Comment (theabmitra) [reply platform-operator/controllers/verrazzano/component/rancherbackup/rancher_backup.go]: done

**Review by michael-cico — DISMISSED** [review]

**Review by michael-cico — APPROVED** [review]

File: platform-operator/controllers/verrazzano/component/rancherbackup/rancher_backup.go [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+	if err != nil {
+		return nil, err
+	}
+
+	bomSubcomponent, err := bomFile.GetSubcomponent("rancher-backup-kubectl")
+	if err != nil {
+		return nil, err
+	}
+
+	rancherRepo := fmt.Sprintf("%s/%s/%s", bomSubcomponent.Registry, bomSubcomponent.Repository, bomSubcomponent.Images[0].ImageName)
```
[the remark this answers is not in the document]
Comment (michael-cico) [reply platform-operator/controllers/verrazzano/component/rancherbackup/rancher_backup.go:53]: e.g.

```
	images, err := bomFile.BuildImageOverrides("rancher-backup-kubectl")
	if err != nil {
		return nil, err
	}
        kvs = append(kvs, images)
```

**Review by michael-cico — APPROVED** [review]

**Pull request merged**